intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordNpmExecutionListener.kt

```diff
@@ -0,0 +1,180 @@
+				val interpreter = mutRunSettings.interpreter
+
+				val patchedInterpreterPath = patchSip(wslDistribution, interpreter.toString())
+
+				val patchedInterpreter = interpreter.javaClass.getConstructor(Class.forName("java.lang.String")).newInstance(patchedInterpreterPath)
+
+				val toInterpreterRef =  patchedInterpreter.javaClass.getMethod("toRef")
+				mutRunSettings.interpreterRef = toInterpreterRef.invoke(patchedInterpreter)
+
+				originPackageManagerPackageRef = mutRunSettings.packageManagerPackageRef
+
+				mutRunSettings.packageManager?.let {
+					packageManager -> {
+						val getSystemIndependentPath = packageManager.javaClass.getMethod("getSystemIndependentPath")
+						val packageManagerPath = getSystemIndependentPath.invoke(packageManager) as String
+
+						val patchedPath = patchSip(wslDistribution, packageManagerPath)
+
+						val patchedPackageManager = packageManager.javaClass.getConstructor(Class.forName("java.lang.String")).newInstance(patchedPath)
+
+						val createPackageManagerPackageRef = originPackageManagerPackageRef!!.javaClass.methods.find { m -> m.name == "create" && m.parameterTypes[0].name != "java.lang.String" }
+
+						mutRunSettings.packageManagerPackageRef = createPackageManagerPackageRef!!.invoke(null, patchedPackageManager)
+					}
+				}
+			}
+		} catch (e: Exception) {
+			MirrordNotifier.notify(
+					"${runProfile::class.qualifiedName}: $e",
+					NotificationType.ERROR,
+					env.project
+			)
+		}
+	}
+
+	private fun clearNpmEnv(env: ExecutionEnvironment) {
+		if (originInterpreterRef == null && originPackageManagerPackageRef == null && mirrordEnv.isEmpty()) {
+			return
+		}
+
+		val runProfile = env.runProfile
+
+		try {
+			val getRunSettings = runProfile.javaClass.getMethod("getRunSettings")
+			val runSettings = getRunSettings.invoke(runProfile)
+
+			val getEnvData = runSettings.javaClass.getMethod("getEnvData")
+			val envData = getEnvData.invoke(runSettings) as EnvironmentVariablesData
+
+			val newEnvData = envData.with(originEnv)
+			originEnv.clear()
+
+			val toBuilder = runSettings.javaClass.getMethod("toBuilder")
+			val builder = toBuilder.invoke(runSettings)
+
+			val setEnvData = builder.javaClass.getMethod("setEnvData", newEnvData.javaClass)
+			setEnvData.invoke(builder, newEnvData)
+
+			if (SystemInfo.isMac) {
+				builder.javaClass.getMethod("setInterpreterRef", originInterpreterRef!!.javaClass).invoke(builder, originInterpreterRef)
+				originInterpreterRef = null
+
+				builder.javaClass.getMethod("setPackageManagerPackageRef", originPackageManagerPackageRef!!.javaClass).invoke(builder, originPackageManagerPackageRef)
+				originPackageManagerPackageRef = null
+			}
+
+			val build = builder.javaClass.getMethod("build")
+			val newRunSettings = build.invoke(builder)
+
+			val setRunSettings = runProfile.javaClass.getMethod("setRunSettings", newRunSettings.javaClass)
+			setRunSettings.invoke(runProfile, newRunSettings)
+		} catch (e: Exception) {
+			MirrordNotifier.notify(
```

Comment:
handle better: log the error and say internal error, please report or something similar.